[MRG] Fixes #11441 adding comments to clarify Spectral Clustering and spectral_embedding

benbo opened the pull request:
Added clarifying comments to spectral.py and spectral_embedding_.py on use of normalized Laplacian and the exact eigenproblem being solved. 

@GaelVaroquaux would you mind checking if you approve of the changes?

benbo committed:
Update comments in spectral.py

Clarified what variable maps corresponds to.

benbo committed:
Update comments in spectral_embedding_.py

Update comments in spectral_embedding_.py to clarify laplacian normalization and use of initial approximation of eigenvectors for lobpcg.

benbo committed:
Update comments in spectral.py

Added comment to clarify exact eigenproblem that is solved.

benbo committed:
Fix formatting spectral.py

Deleted line break.

benbo committed:
Update spectral.py

remove trailing whitespace

benbo committed:
Update spectral_embedding_.py

Removed trailing whitespaces

glemaitre commented on sklearn/manifold/spectral_embedding_.py:
```diff
@@ -181,7 +181,7 @@ def spectral_embedding(adjacency, n_components=8, eigen_solver=None,
 
     norm_laplacian : bool, optional, default=True
-        If True, then compute normalized Laplacian.
+        If True, then compute symmetric normalized Laplacian.
```
Not sure that this is pertinent since the parameter does not influence the symmetry.

benbo replied:
That is true. However, it is unclear otherwise if the algorithm uses the symmetric normalized Laplacian or the random walk normalized Laplacian. Do you have a suggestion for a better spot somewhere to mention this?

glemaitre commented on sklearn/manifold/spectral_embedding_.py:
```diff
@@ -291,12 +295,14 @@ def spectral_embedding(adjacency, n_components=8, eigen_solver=None,
         laplacian = _set_diag(laplacian, 1, norm_laplacian)
         ml = smoothed_aggregation_solver(check_array(laplacian, 'csr'))
         M = ml.aspreconditioner()
+        # Create initial approximation X to eigenvectors
```
I would also remove this comment

benbo replied:
Anyone trying to understand what is being done here would have to call up the LOBPCG documentation. If comments should generally be this sparse in sklearn then I can certainly remove it. But I honestly think the code could use the explanations.

glemaitre commented on sklearn/manifold/spectral_embedding_.py:
```diff
@@ -236,6 +236,9 @@ def spectral_embedding(adjacency, n_components=8, eigen_solver=None,
         warnings.warn("Graph is not fully connected, spectral embedding"
                       " may not work as expected.")
 
+    # When norm_laplacian=True csgraph_laplacian returns the symmetric
```
I would remove this comment since this indicated in the documentation of `csgraph_laplacian`

benbo replied:
The documentation of csgraph_laplacian does not mention that it is the symmetric normalization. This is important for the algorithm, that's why I added it here. However, I did create a scipy issue to address this so it could be removed here.

benbo replied:
I removed the comment

benbo committed:
Remove comment describing csgraph_laplacian 

Remove comment describing csgraph_laplacian return variables when norm_laplacian=True
